pyam/utils.py

```diff
@@ -208,24 +208,46 @@ def style_df(df, style='heatmap'):
         return df.style.background_gradient(cmap=cm)
 
 
-def find_depth(data, s, level):
-    # determine function for finding depth level =, >=, <= |s
+def find_depth(data, s='', level=None):
+    """
+    return or assert the depth (number of `|`) of variables
+
+    Parameters
+    ----------
+    data : pd.Series of strings
+        IAMC-style variables
+    s : str, default ''
+        remove leading `s` from any variable in `data`
+    level : int or str, default None
+        if None, return depth (number of `|`); else, return list of booleans
+        whether depth satisfies the condition (equality if `level` is int,
+        >= if `.+`,  <= if `.-`)
+    """
+    # remove wildcard as last character from string, find depth
+    s = s.rstrip('*')
+    regexp = re.compile('^' + _escape_regexp(s))
+
+    def _pipe(val):
+        return len(re.compile('\\|').findall(re.sub(regexp, '', val))) \
+            if regexp.match(val) else None
+
+    # if no level test is specified, return the depth as int
+    if level is None:
+        return list(map(lambda x: _pipe(x), data))
+
+    # if `level` is given, set function for finding depth level =, >=, <= |s
     if not isstr(level):
-        test = lambda x: level == x
+        test = lambda x: level == x if x is not None else False
     elif level[-1] == '-':
         level = int(level[:-1])
-        test = lambda x: level >= x
+        test = lambda x: level >= x if x is not None else False
     elif level[-1] == '+':
         level = int(level[:-1])
-        test = lambda x: level <= x
+        test = lambda x: level <= x if x is not None else False
     else:
-        raise ValueError('Unknown level type: {}'.format(level))
+        raise ValueError('Unknown level type: `{}`'.format(level))
 
-    # determine depth
-    pipe = re.compile('\\|')
-    regexp = str(s).replace('*', '')
-    apply_test = lambda val: test(len(pipe.findall(val.replace(regexp, ''))))
-    return list(map(apply_test, data))
+    return list(map(lambda x: test(_pipe(x)), data))
```

Comment:
probably cleaner to include the `_pipe` calls above in lambda defs and just have here:

```
return list(map(test, data))
```

Reply:
done